PowerEditor/src/WinControls/DockingWnd/DockingManager.cpp

```diff
@@ -968,12 +970,21 @@ int DockingManager::FindEmptyContainer()
 
         for (size_t iTb = 0, len = vTbData.size(); iTb < len; ++iTb)
         {
-            pArrayPos[vTbData[iTb]->iPrevCont] = TRUE;
+			if ((vTbData[iTb]->iPrevCont < static_cast<int>(dockingContVectorSize)) && (vTbData[iTb]->iPrevCont >= -1))
+			{
+				pArrayPos[vTbData[iTb]->iPrevCont] = TRUE;
+			}
+			else
+			{
+				// ? invalid config.xml input data
+				assert((vTbData[iTb]->iPrevCont < static_cast<int>(dockingContVectorSize)) && (vTbData[iTb]->iPrevCont >= -1));

```

Comment:
Is assert necessary?

Reply:
I would prefer to have this assert there so that one can at least in the Debug build be notified that the current config.xml data for the Docking Manager is not quite right.